src/memray/_memray/source.cpp

```diff
@@ -56,13 +66,13 @@ FileSource::close()
 void
 FileSource::_close()
 {
-    d_stream.close();
+    d_raw_stream.close();
```

Comment:
Shouldn't this be `d_stream->close()`?

Reply:
It seems strange to me that you're closing the raw stream but leaving `d_stream` alone...

Reply:
AFAIK the lz4 stream doesn't have `close()` and just acts as a facade to the other stream

Reply:
OK. That's annoying, but I'm gonna leave it the way you have it rather than introduce more complexity here.